### PR TITLE
feat(go): support basic pkg

### PR DESCRIPTION
**Pull request by Reanon**
1. add list_node & tree_node

### CODE REVIEW

File: codes/go/pkg/list_node.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// NewListNode Generate a list node with an val
+func NewListNode(v int) *ListNode {
+	return &ListNode{
+		Next: nil,
+		Val:  v,
+	}
+}
+
+// ArrayToLinkedListLinkedList Generate a linked list with an array
+func ArrayToLinkedListLinkedList(arr []int) *ListNode {
```
Comment (krahets) [inline codes/go/pkg/list_node.go:28]: Is this `ArrayToLinkedList` ?

**Review by krahets — COMMENTED** [review]
Thanks for the great work! There is a comment that need to be addressed, thanks!

File: codes/go/pkg/list_node.go (hunk 1 above)
Comment (Reanon) [reply codes/go/pkg/list_node.go]: Amended in next commit.

**Review by krahets — APPROVED** [review]
Thanks! Now merge it.
#16

**Pull request merged**